### PR TITLE
spelling corrections etc

### PR DESCRIPTION
**Pull request by PublicHealthDataGeek**
some spelling corrections, some comments for RL to consider, some definitions added for clarity, some comments added to code.

### CODE REVIEW

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd:12]: This is definitely much better. 'Tapping' is a funny word, would 'typing Enter' be better?... Probably not.

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd:23]: That is a very good point you make about `##`.

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd:39]: Thanks for this, definite improvement in terms of readability. 'they assign contents' is fine.

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd:51]: :+1:

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd:68]: Important clarification.

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd:70]: How about:

```
In plain English, things that go inside the curved brackets following a function name are the function's *arguments*.
```

Even more specific.

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd]: Good point. How about just 'Subsetting'? 

Would

'Subsetting vectors' 

be any better?

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd]: Good point. How about:

```
The most basic type of R object is a *vector*, a sequence of values of the same type.
`x`, `y` and `z` were all *numeric vectors* with a length of 1; `casualty_type` is a *character vector* (because it contains letters that cannot be added) or length 3; `casualty_age` is a *numeric vector* of length three.
In R, subsets of vectors can be returned by providing numbers representing the positions of the elements we want (e.g. '2' representing selection of the 2^nd^ element) or with logical (`TRUE` or `FALSE`) values associated with the elements that should be selected. 
```

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd:220]: :+1:

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd]: How about:

```
 There are many ways to recode values in R, the simplest and most mature of which is the use of factors, which are whole numbers representing characters, as shown below: 
```

On second thoughts, is this even a useful example? It is a bit confusing I think so it may be worth scrapping it IMO...

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd]: Yes that's the right term. How about the following?

``` 
You can also save individual R objects in the Rds format. The Rds format is R's native data format, meaning that any R object can be saved as an Rds file.
```

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd:297]: :+1:

**Review by Robinlovelace — CHANGES_REQUESTED** [review]
Great set of suggestions. I've suggested a few tweaks before merging. Thanks Caroline!

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:12]: Yes it would

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:23]: not sure if I need to comment or not to this...Just my lack of knowledge on how to use github again!

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:39]: :+1:

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:68]: :)

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:51]: :)

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:70]: Yes - even better!

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: Would need to define vector which I mentioned below. If 2.3 was subsetting then would you merge it with 2.4 (which is also subsetting)?  I think you could keep as subsetting by index or name if you make it clearer in the text what 'index' and 'name' mean.

Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: ah looks like you have sorted it below!!!

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: Few more edits below- hope they make sense 


The most basic type of R object is a *vector*, a sequence of values of the same type, FOR EXAMPLE, TYPE COULD BE NUMERIC.
`x`, `y` and `z` were all *numeric vectors* with a length of 1; `casualty_type` is a *character vector* (because it contains letters that cannot be added) OF length 3; `casualty_age` is a *numeric vector* of length 3. CHANGED THREE TO 3 FOR COSNISTENCY
In R, subsets of vectors can be returned by providing numbers representing the positions of the elements we want WITHIN THE VECTOR (e.g. '2' representing selection of the 2^nd^ element) or with logical (`TRUE` or `FALSE`) values associated with the elements that should be selected.

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: There are quite a few use cases for using factors in STATS19 eg recoding vehicles to simpler categories so I think it is helpful to have a look at factors.  I think the wording you have suggested is better - perhaps worth saying something that factors are usually used for categorical data - i think that would help - would a male/femae/other example would be more helpful?

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: Perhaps you could reference it to Excel syaing that 'in the same way that .xls is the native Excel data format.  ?

Comment (Robinlovelace) [reply 02-basics.Rmd]: good idea

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:23]: :)

**Comment by Robinlovelace** [comment]
Thanks for all the replies Caroline, it seems we're on the same wavelength. Do you want to have one more go at changing a few things? If so let me know when you're done (you can edit this version, let me know if you have any issues finding out how) and I'll merge this, can always tidy-up after :broom:

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: SUGGESTED WORDING CHANGES BELOW

The most basic type of R object is a *vector*, a sequence of values of the same type FOR EXAMPLE, A SET OF NUMBER.
`x`, `y` and `z` were all *numeric vectors* with a length of 1; `casualty_type` is a *character vector* (because it contains letters that cannot be added) OF length 3; `casualty_age` is a *numeric vector* of length 3(CHANGED TO THREE FOR CONSISTENCY).


In R, subsets of vectors can be returned by providing numbers representing the positions of the elements IN THE VECTOR  we want (e.g. '2' representing selection of the 2^nd^ element) or with logical (`TRUE` or `FALSE`) values associated with the elements that should be selected.

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: I think the wording is now good.  A better example might be to use gender e.g male, female, other as that is more relevent and highlights how we tend to use factors - to code categorical data.  Perhaps add a sentence saying that are usually used for categorical variables.

File: 02-basics.Rmd
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: Suggest add the wording below or similar:  
This is comparable to Excel where the native data format is xls.

**Comment by PublicHealthDataGeek** [comment]
Hi Robin, thanks for the feedback.  I dont know what quite happened - I put in various replies then looked at help and thought that I had lost all my comments but seeing your reply 5 mins ago it looks like you might have got them...?  anyway, I had gone through again and done them again and so you might get them again.  I am not sure how to 'changes things and edit this version'?  Do I just refresh the book and then go through and edit again?

**Comment by Robinlovelace** [comment]
Hi @PublicHealthDataGeek I can see all of your comments but some of them show as 'resolved' if you edit sections as shown below, nothing has been lost.

image

**Comment by Robinlovelace** [comment]
Can you try making a few more edits here @PublicHealthDataGeek?: https://github.com/PublicHealthDataGeek/rrsrr/edit/patch-3/02-basics.Rmd?pr=%2Fsaferactive%2Frrsrr%2Fpull%2F6

Partly for interest and partly to make the editing process quicker when I merge this PR.

**Comment by PublicHealthDataGeek** [comment]
HI Robin - used the link - it has posted it under your comment yestereday.  Then I clicked on 'request review'above so fingers crossed it all links seemlessly and you can see the futher changes I made to incorporate our discussion.....

File: 02-basics.Rmd
Comment (Robinlovelace) [inline 02-basics.Rmd:281]: Many thanks for the update @PublicHealthDataGeek !

**Comment by Robinlovelace** [comment]
Great news @PublicHealthDataGeek it works, this is what I see: 

image

**Review by Robinlovelace — APPROVED** [review]
Many thanks for the updates Caroline, way better.

**Comment by Robinlovelace** [comment]
Good practice:

image

**Comment by Robinlovelace** [comment]
Heads-up @PublicHealthDataGeek I've just cleaned up Chapter 2. Please take a look at the new shiny results here: https://saferactive.github.io/rrsrr/basics.html

**Pull request merged**
